### ISSUE TITLE
Is it possible to avoid manually capturing a backtrace when variant has no `source` error?

### ISSUE THREAD

**Issue by casey** [shepmaster/snafu#457]
Using this error type as an example:

```rust
#[derive(Snafu)]
enum Error {
  Foo { backtrace: Backtrace, source: std::io::Error },
  Bar { backtrace: Backtrace },
}
```

When using a context selector, it's possible to automatically capture a backtrace:

```rust
read_file().context(FooSnafu)?;
```

Is it possible to avoid manually capturing a backtrace when there is no source error?

```rust
if !condition {
  // currently I do this:
  return Err(Error::Bar { backtrace: Backtrace });
 
  // but would like to be able to do something like this:
  return Err(BarSnafu.into());
}
```

PS As always, thanks for `snafu`, it's rad.

**Comment by shepmaster** [comment]
Yep. See [the docs](https://docs.rs/snafu/latest/snafu/guide/what_code_is_generated/index.html#generated-code) (search for "Otherwise, the context selector will have the inherent methods")

TL;DR: try this:

```rust
if !condition {
  return BarSnafu.fail();
}
```

Or use [`ensure`](https://docs.rs/snafu/latest/snafu/macro.ensure.html):

```rust
ensure!(condition, BarSnafu);
```

**Comment by casey** [comment]
Awesome, that's perfect. Thank you!
